File: CS-Examples/25_TrackChanges/AcceptOrRejectTrackedChanges.vb
Imports Spire.Xls

Namespace AcceptOrRejectTrackedChanges
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file named "TrackChanges.xlsx" from a specific path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\TrackChanges.xlsx")

            ' Reject all tracked changes in the workbook
            workbook.RejectAllTrackedChanges()

            ' Specify the output file name as "AcceptOrRejectTrackedChanges.xlsx"
            Dim outputFile As String = "AcceptOrRejectTrackedChanges.xlsx"

            ' Save the workbook to the specified output file, using the Version 2013 file format
            workbook.SaveToFile(outputFile, FileFormat.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(outputFile)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/25_TrackChanges/AcceptOrRejectTrackedChanges.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace AcceptOrRejectTrackedChanges
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\TrackChanges.xlsx");

            // Accept the changes or reject the changes.
            //workbook.AcceptAllTrackedChanges();
            workbook.RejectAllTrackedChanges();

            // Save to file.
            string outputFile = "AcceptOrRejectTrackedChanges.xlsx";       
            workbook.SaveToFile(outputFile, FileFormat.Version2013);

            // Dispose of the workbook object to release resources 
            workbook.Dispose();

            //View the document
            FileViewer(outputFile);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
